File: invesdwin-scripting-parent/invesdwin-scripting-rust/invesdwin-scripting-rust-runtime-contract/src/test/java/de/invesdwin/scripting/rust/runtime/contract/callback/ParametersAndReturnsTestDecimal.rs
println!("getDecimal");
let getDecimal: f64 = callback("getDecimal", &[]);
println!("getDecimal: {}", getDecimal);
callback_void("setDecimal", &[param(getDecimal)]);

println!("getDecimalVector");
let getDecimalVector: Vec<f64> = callback("getDecimalVector", &[]);
println!("getDecimalVector: {:?}", getDecimalVector);
callback_void("setDecimalVector", &[param(getDecimalVector)]);

println!("getDecimalVectorAsList");
let getDecimalVectorAsList: Vec<f64> = callback("getDecimalVectorAsList", &[]);
println!("getDecimalVectorAsList: {:?}", getDecimalVectorAsList);
callback_void("setDecimalVectorAsList", &[param(getDecimalVectorAsList)]);

println!("getDecimalMatrix");
let getDecimalMatrix: Vec<Vec<f64>> = callback("getDecimalMatrix", &[]);
println!("getDecimalMatrix: {:?}", getDecimalMatrix);
callback_void("setDecimalMatrix", &[param(getDecimalMatrix)]);

println!("getDecimalMatrixAsList");
let getDecimalMatrixAsList: Vec<Vec<f64>> = callback("getDecimalMatrixAsList", &[]);
println!("getDecimalMatrixAsList: {:?}", getDecimalMatrixAsList);
callback_void("setDecimalMatrixAsList", &[param(getDecimalMatrixAsList)]);

File: invesdwin-scripting-parent/invesdwin-scripting-haskell/invesdwin-scripting-haskell-runtime-ghci/src/test/java/de/invesdwin/scripting/haskell/runtime/callback/ParametersAndReturnsTestDecimal.hs
putStrLn ( show ( "getDecimal" ) )
getDecimal :: Double <- callback "getDecimal" []
putStrLn ( show ( getDecimal ) )
callback "setDecimal" [param getDecimal] :: IO ()

putStrLn ( show ( "getDecimalVector" ) )
getDecimalVector :: [Double] <- callback "getDecimalVector" []
putStrLn ( show ( getDecimalVector ) )
callback "setDecimalVector" [param getDecimalVector] :: IO ()

putStrLn ( show ( "getDecimalVectorAsList" ) )
getDecimalVectorAsList :: [Double] <- callback "getDecimalVectorAsList" []
putStrLn ( show ( getDecimalVectorAsList ) )
callback "setDecimalVectorAsList" [param getDecimalVectorAsList] :: IO ()

putStrLn ( show ( "getDecimalMatrix" ) )
getDecimalMatrix :: [[Double]] <- callback "getDecimalMatrix" []
putStrLn ( show ( getDecimalMatrix ) )
callback "setDecimalMatrix" [param getDecimalMatrix] :: IO ()

putStrLn ( show ( "getDecimalMatrixAsList" ) )
getDecimalMatrixAsList :: [[Double]] <- callback "getDecimalMatrixAsList" []
putStrLn ( show ( getDecimalMatrixAsList ) )
callback "setDecimalMatrixAsList" [param getDecimalMatrixAsList] :: IO ()
